src/strands/agent/interrupt.py

```diff
@@ -0,0 +1,59 @@
+"""Track the state of interrupt events raised by the user for human-in-the-loop workflows."""
+
+from dataclasses import asdict, dataclass, field
+from typing import Any
+
+from ..interrupt import Interrupt
+
+
+@dataclass
+class InterruptState:
+    """Track the state of interrupt events raised by the user.
+
+    Note, interrupt state is cleared after resuming.
+
+    Attributes:
+        interrupts: Interrupts raised by the user.
+        context: Additional context associated with an interrupt event.
+        activated: True if agent is in an interrupt state, False otherwise.
+    """
+
+    interrupts: dict[str, Interrupt] = field(default_factory=dict)
+    context: dict[str, Any] = field(default_factory=dict)
+    activated: bool = False
+
+    def activate(self, context: dict[str, Any] | None = None) -> None:
+        """Activate the interrupt state.
+
+        Args:
+            context: Context associated with the interrupt event.
+        """
+        self.context = context or {}
+        self.activated = True
+
+    def deactivate(self) -> None:
+        """Deacitvate the interrupt state.
+
+        Interrupts and context are cleared.
+        """
+        self.interrupts = {}
+        self.context = {}
+        self.activated = False
```

Comment:
Is `activated == bool(self.context)` supposed to always be true? (assuming [that empty dictionaries are falsey](https://stackoverflow.com/a/23177452/548304))

If so, can we turn activated into a getter and not store it in the dictionary?